src/esp/gfx/CubeMap.cpp

```diff
@@ -0,0 +1,399 @@
+                          const std::string& imageFilePrefix,
+                          const std::string& imageFileExtension) {
+  auto loadImporter = [&](const std::string& plugin) {
+    Cr::PluginManager::Manager<Mn::Trade::AbstractImporter> manager;
+    Cr::Containers::Pointer<Mn::Trade::AbstractImporter> importer =
+        manager.loadAndInstantiate(plugin);
+    CORRADE_ASSERT(importer,
+                   "CubeMap::loadTexture(): cannot initialize" << plugin, );
+
+    imageImporterManger_.set<Mn::Trade::AbstractImporter>(
+        plugin, importer.release(), Mn::ResourceDataState::Final,
+        Mn::ResourcePolicy::Manual);
+  };
+
+  std::string plugin;
+  if (imageFileExtension == "png") {
+    plugin = "PngImporter";
+  } else if (imageFileExtension == "jpg") {
+    plugin = "JpegImporter";
+  } else if (imageFileExtension == "hdr") {
+    plugin = "HdrImporter";
+  } else {
+    LOG(INFO) << "CubeMap::loadTexture(): cannot support image type "
+              << imageFileExtension;
+    CORRADE_INTERNAL_ASSERT_UNREACHABLE();
+  }
+
+  Mn::Resource<Mn::Trade::AbstractImporter> importer =
+      imageImporterManger_.get<Mn::Trade::AbstractImporter>(plugin);
+  // create one if there is no such importer
+  if (!importer) {
+    loadImporter(plugin);
+  }
+  CORRADE_INTERNAL_ASSERT(importer);
+
+  std::string coordStrings[6] = {".+X", ".-X", ".+Y", ".-Y", ".+Z", ".-Z"};
+  int imageSize = 0;
+  // set images
+  Mn::GL::CubeMapTexture* texture;
+  switch (type) {
+    case TextureType::Color:
+      texture = textures_[TextureType::Color].get();
+      break;
+
+    case TextureType::Depth:
+      texture = textures_[TextureType::Depth].get();
+      break;
+
+    default:
+      CORRADE_INTERNAL_ASSERT_UNREACHABLE();
+      break;
+  }
+  for (int iFace = 0; iFace < 6; ++iFace) {
+    // open image file
+    std::string filename =
+        imageFilePrefix + getTextureTypeFilenameString(type) +
+        coordStrings[iFace] + "." + std::string{imageFileExtension};
+    importer->openFile(filename);
+    Cr::Containers::Optional<Mn::Trade::ImageData2D> image =
+        importer->image2D(0);
+
+    // sanity checks
+    CORRADE_INTERNAL_ASSERT(image);
+    Mn::Vector2i size = image->size();
+    CORRADE_ASSERT(
+        size.x() == size.y(),
+        " CubeMap::loadTexture(): each texture image must be a square.", );
+    if (iFace == 0) {
+      imageSize = size.x();
+      reset(imageSize);
+    } else {
+      CORRADE_ASSERT(size.x() == imageSize,
+                     " CubeMap::loadTexture(): texture images must have the "
+                     "same size.", );
+    }
+
+    texture->setSubImage(convertFaceIndexToCubeMapCoordinate(iFace), 0, {},
+                         *image);
+  }
+  // Color texture ONLY, NOT for depth
+  if (flags_ & (Flag::BuildMipMap | Flag::ColorTexture)) {
+    texture->generateMipmap();
+  }
+  // We don't need the importer anymore
+  imageImporterManger_.free<Mn::Trade::AbstractImporter>();
```

Comment:
Why do you need a ResourceManager at all, then, if the plugin doesn't even outlive the function it was instantiated in? It could be just a function-local variable.